lib/util/state.js

```diff
@@ -304,6 +304,103 @@ export function getActiveItinerary (otpState) {
     : null
 }
 
+/**
+ * Returns true if the OTP state contains a response within the current active search.
+ * @param {*} state The entire redux state used to obtain the response.
+ */
+function hasResponse (state) {
+  const activeSearch = getActiveSearch(state.otp)
+  return !!(activeSearch && activeSearch.response)
+}
+
+/**
+ * Returns true if the OTP state contains one or more OTP responses for the current active search.
+ * @param {*} state The entire redux state used to obtain the responses.
+ */
+function hasAtLeastOneResponse (state) {
+  const activeSearch = getActiveSearch(state.otp)
+  return activeSearch && activeSearch.response && activeSearch.response.length > 0
+}
+
+/**
+ * Returns the raw OTP response for the current active search.
+ * @param {*} state The entire redux state used to obtain the response.
+ */
+function getOtpResponse (state) {
+  const activeSearch = getActiveSearch(state.otp)
+  return activeSearch && activeSearch.response && activeSearch.response.otp
```

Comment:
Lots of optional chaining opportunities everywhere in this file! E.g.:
```suggestion
  return getActiveSearch(state.otp)?.response?.otp
```

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/366/commits/344fc9fb80b4de8724aa55b4bfae5dafd15d0fc9.